renku/core/util/os.py

```diff
@@ -165,32 +193,38 @@ def delete_file(filepath: Union[Path, str], ignore_errors: bool = True, follow_s
             pass
 
 
-def hash_file(path: Union[Path, str]) -> Optional[str]:
-    """Calculate the sha256 hash of a file."""
+def hash_file(path: Union[Path, str], hash_type: str = "sha256") -> Optional[str]:
+    """Hash a file."""
+    hash_type = hash_type.lower()
+    assert hash_type in ("sha256", "md5")
+
     if not os.path.exists(path):
         return None
 
-    sha256_hash = hashlib.sha256()
+    hash_value = hashlib.sha256() if hash_type == "sha256" else hashlib.md5()
 
     with open(path, "rb") as f:
         for byte_block in iter(lambda: f.read(BLOCK_SIZE), b""):
-            sha256_hash.update(byte_block)
+            hash_value.update(byte_block)
+
+    return hash_value.hexdigest()
 
-    return sha256_hash.hexdigest()
 
+def hash_str(content: str, hash_type: str = "sha256"):
```

Comment:
I wonder if we could unify these two functions using StringIO for strings and `open()` for files? As in, have `hash_str` call `hash_file` or both call the same helper method?